### PR TITLE
Update push token on login to allow multiple users on mobile devices

### PR DESCRIPTION
**Pull request by trmartin4**
https://bitwarden.atlassian.net/browse/SG-816
https://bitwarden.atlassian.net/browse/SG-814
https://bitwarden.atlassian.net/browse/SG-823

## Type of change

```
- [X] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
We are experiencing a problem today where push notifications are not being sent to mobile devices when there are multiple users logged in. 

There is still a scenario where no push token would get set for a given user + device combination, and that would be if a user logs in to a device and gets an access token prior to the PNS providing a push token to the device.  However, if such a scenario occurs, the user can log out and log back in with that user and the token would be updated on the subsequent authentication request.

## Code changes

**BaseRequestValidator.cs:**
- When request for a token comes to Identity, we used to only call `CreateOrUpdateRegistrationAsync()` (as a part of `SaveAsync()` on the `DeviceService`) **when it was a new device**.  This meant that subsequent login requests never updated the push token. 
- I changed this so regardless of whether the device is new **or** existing, every login request will attempt to set the push token for that logged-in user in the Azure Notification Hub.

**AuthRequestsController:**
- Added `userId` to lookup for the correct device, in case there are multiple logged in

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Api/Utilities/DeviceUpdateHandler.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+namespace Bit.Api.Utilities;
+
+
+public class DeviceUpdateHandler
+{
+
+    public IDeviceService _deviceService;
+    public IPushRegistrationService _pushRegistrationService;
+
+}
```
Comment (differsthecat) [inline src/Api/Utilities/DeviceUpdateHandler.cs:12]: Where is this used? I can't find the reference to it

Comment (trmartin4) [reply src/Api/Utilities/DeviceUpdateHandler.cs:12]: Oops.  I deleted the file from the solution but not the hard drive.  I'll clean it up.

File: src/Core/Services/Implementations/DeviceService.cs [hunk 2]
```diff
@@ -16,20 +16,20 @@ public class DeviceService : IDeviceService
...
             await _deviceRepository.ReplaceAsync(device);
+            result = device;
         }
-
-        await _pushRegistrationService.CreateOrUpdateRegistrationAsync(device.PushToken, device.Id.ToString(),
-            device.UserId.ToString(), device.Identifier, device.Type);
+        return result;
     }
 
     public async Task ClearTokenAsync(Device device)
```
Comment (differsthecat) [inline src/Core/Services/Implementations/DeviceService.cs:35]: It looks like with the removal of the `/clear-token`, this isn't used anymore. Is that correct? If so, should we remove this method and definition in IDeviceService?

Comment (trmartin4) [reply src/Core/Services/Implementations/DeviceService.cs]: Yes, thank you!  I just updated with another commit.

**Review by differsthecat — APPROVED** [review]

**Pull request merged**